geom/geom/src/TGeoVoxelFinder.cxx

```diff
@@ -1687,6 +1687,11 @@ void TGeoVoxelFinder::SortAll(Option_t *)
       boundaries[2*id+4*nd] = fBoxes[6*id+5]-fBoxes[6*id+2];
       boundaries[2*id+4*nd+1] = fBoxes[6*id+5]+fBoxes[6*id+2];
    }
+   if (static_cast<size_t>(nmaxslices)*nperslice > static_cast<size_t>(nmaxslices*nperslice)) {
```

Comment:
I'm not sure this detects all types of overflows, such as:
```
root [0] int nmaxslices = 12345678;
root [1] int nperslice = 123456;
root [3] static_cast<size_t>(nmaxslices)*nperslice -> (unsigned long) 1524148023168
root [4] static_cast<size_t>(nmaxslices*nperslice) -> (unsigned long) 1524148023168
```
however, you can store the product in a variable and check if `nmaxslices == prod/nperslice`:
```
root [4] int prod = nmaxslices*nperslice -> (int) 260846532
root [5] prod/nperslice ->(int) 2112
```

Reply:
Should i just cast the right side of the comparison to Int_t instead?

Reply:
Rather try:
```
auto prod = nmaxslices*nperslice;
if (nmaxslices != prod / nperslice) {
 Error(...);
}
```
comparing directly `nmaxslices` with nmaxslices*nperslice/nperslice may not work because the intermediate truncation forced by the store may not happen.